vault/diagnose/helpers.go

```diff
@@ -93,7 +93,7 @@ func (s *Session) Finalize(ctx context.Context) *Result {
 }
 
 // StartSpan starts a "diagnose" span, which is really just an OpenTelemetry Tracing span.
-func StartSpan(ctx context.Context, spanName string, options ...trace.SpanOption) (context.Context, trace.Span) {
+func StartSpan(ctx context.Context, spanName string, options ...trace.SpanStartOption) (context.Context, trace.Span) {
```

Comment:
Not 100% sure of the impact of this change. Seems reasonable and looks like it's mainly used for testing, but I could have missed something.

Reply:
For further context, this change is required because the function signature for `session.tracer.Start` has been updated to accept a `SpanStartOption` instead of a `SpanOption`

Reply:
This looks okay to me. The underlying interface for `SpanStartOption` and `SpanEndOption` are the same. In v0.20.0, they were just placed into a single `SpanOption` like:
```
// SpanOption are options that can be used at both the beginning and end of a span.
type SpanOption interface {
	SpanStartOption
	SpanEndOption
}
```